### ISSUE TITLE
Default case in switch statements not handled correctly

### ISSUE THREAD

**Issue by kelloggm** [typetools/checker-framework#1180]
Consider a switch statement:

```java
switch(x) {
  case 0:
  case 1:
      // do some code...
      break;
   //... more cases
  default:
      // some code
}
```

When the CFG is built, a series of `CaseNode` objects are built for each case - except the default case. This happens in `org.checkerframework.dataflow.cfg.CFGBuilder.CFGTranslationPhaseOne.SwitchBuilder`. In Java, the semantics of the `default` statement should be that it is executed if either: it is reached via fall-through from one of the previous cases, or if none of the previous cases were triggered.

With no `CaseNode` for default, it is impossible to refine a type in the default case based on what came before. See typetools#1172 for some test cases that demonstrate this (and more exposition on the motivation for desiring this change).
